src/pdfnorm.jl

```diff
@@ -0,0 +1,58 @@
+export pdfL2norm
+
+"""
+    pdfL2norm(d::Distribution)
+
+Return the L2 norm of the probability distribution function `f(x)` of the distribution `d`:
+
+```math
+\\int_{S} f(x)^{2} \\mathrm{d} x
+```
+
+where `S` is the support of `f(x)`.
+"""
+pdfL2norm
+
+pdfL2norm(d::Distribution) = throw(ArgumentError("L2 norm not implemented for $(d)"))
+
+function pdfL2norm(d::Beta{T}) where T
```

Comment:
IMO it would be good if one could avoid depending on the type parameter `T` as a heuristic, here and in the other functions, and instead compute the result as much as possible and then just return `oftype(result, Inf)` if needed.

Reply:
Can I ask you why?  Because the parameter is `T<:Real` so not necessarily a floating point?  If so, can use `float(T)` as type.  Here and in the other functions the integral diverges after a certain cut-off, and it would have a completely different analytic expression (if any), so there isn't much result to compute at all.  For example, for the `Chi` distribution
```julia
julia> using SpecialFunctions

julia> f(ν) = (2 ^ (1 - ν) * gamma((2 * ν - 1) / 2)) / gamma(ν / 2) ^ 2
f (generic function with 1 method)

julia> f(0.5)
Inf

julia> f(0.25)
-0.14523516581055995
```
Extending the analytic solution to `ν < 0.5` doesn't make sense, you get negative values.

Reply:
Ok, I thought you meant "compute as much meaningful result as possible in each branch", instead you're saying "always compute the result for the converging branch, and maybe return `Inf` if necessary disregarding the quantity computed before" :wink: